packages/eslint-plugin/src/rules/no-unsafe-enum-comparison.ts

```diff
@@ -100,11 +123,44 @@ export default util.createRule({
           }
         }
 
-        if (
-          typeViolates(leftTypeParts, right) ||
-          typeViolates(rightTypeParts, left)
-        ) {
-          context.report({
+        if (typeViolates(leftTypeParts, right)) {
+          const leftEnumKey = getEnumKeyForLiteral(leftEnumTypes, node.right);
+
+          if (leftEnumKey) {
+            // TODO: Add fixer.
+            return context.report({
+              messageId: 'mismatchedSimilar',
+              node,
+              data: {
+                replacement: leftEnumKey,
+              },
+            });
+          }
+
+          return context.report({
+            messageId: 'mismatched',
+            node,
+          });
+        }
+
+        if (typeViolates(rightTypeParts, left)) {
+          const rightEnumKey = getEnumKeyForLiteral(
+            [...rightEnumTypes.values()],
+            node.left,
+          );
+
+          if (rightEnumKey) {
+            // TODO: Add fixer.
+            return context.report({
+              messageId: 'mismatchedSimilar',
+              node,
+              data: {
+                replacement: rightEnumKey,
+              },
+            });
+          }
+
+          return context.report({
```

Comment:
By "suggestion" I meant the actual ESLint feature: https://eslint.org/docs/latest/extend/custom-rules#providing-suggestions. I don't think we need to add any new reports?

Reply:
Oh ok, sorry, didn't get that 

so you mean something like this?
```TS
if (typeViolates(leftTypeParts, right)) {
  const enumKey = getEnumKeyForLiteral(...);

  return context.report({
    messageId: 'mismatched',
    node,
    suggest: [
      {
         desc: "Replace with the literal value with an enum member.",
         fix(fixer) {
           return fixer.replaceText(...);
         }
      },
    ],
  });
}
```

In addition, is there any place where I can find some documentation about this?
My current implementation of `getEnumKeyForLiteral()` is basically me guessing using TS, and some basic knowledge of ASTs, but it's not 100% working and I can't seem to find a good resource about this stuff.

___
(sorry if those are dumb questions, I just watched [this](https://www.youtube.com/watch?v=OVNQWzdhCQA) video and wanted to help 😅)

Reply:
Yeah along those lines! I was thinking vaguely like:

```diff
context.report({
  messageId: 'mismatched',
  node,
+  suggest: [
+    {
+      fix(fixer) {
+        return fixer.replaceText(...);
+      },
+      memessageId "replaceValueWithEnum",
+    },
  ]
});
```

The idea here is that we're not changing when the rule reports. We're only adding in a new `suggest` property to add in a suggestion.

I haven't looked too deeply, but it _might_ get tricky trying to figure out what text to add in. If the enum is declared in another file then I don't know that we have a good way to add in an `import` for it automatically. No worries if you don't implement that part just yet.

...and cool! Really happy to see folks come in from MiTS 😄 welcome!

Reply:
The main reason I split it into two conditions is to determine which of them violated and to replace the text accordingly.
Will try to look at it again and see if I can get any progress